### ISSUE TITLE
Running headed tests on Travis

### ISSUE THREAD

**Issue by ssoloff** [triplea-game/triplea#2173]
This is a follow-up to a [discussion](https://github.com/triplea-game/triplea/commit/4877b1c76705412aca7fcfbe51d484eac2b46455#commitcomment-23345443) begun in #2137.

In #2137, several unit tests were added that cause a Swing EDT to be spun up.  In a headless environment, such as Travis, this causes a `HeadlessException` to be thrown during the execution of these tests.  The workaround employed in #2137 was to ignore the exception and force the test to pass.  For example:

```java
@Test(expected = NullPointerException.class)
public void contentsIsRequired() {
  try {
    JDialogBuilder.builder()
        .title("title")
        .parentFrame(new JFrame())
        .build();
  } catch (final HeadlessException e) {
    // this is okay, we'll see this in travis
    throw new NullPointerException("fake the exception to pass the test");
  }
}
```

In the past, we've avoided this type of problem by hiding the Swing dependencies behind additional abstractions to avoid any interaction that may cause the Swing graphics subsystem to be initialized.  That does not appear to be possible in this case due to the explicit dependency of `JDialogBuilder` on a non-`null` `JFrame` instance.

IMO, I'd prefer not to see this kind of workaround in tests for the following reasons:

1. The additional error handling is distracting to a test reader.  And in the case of the above example, having to throw a specific type of exception to make the test pass (`NullPointerException`) creates additional coupling to the implementation and potentially requires more maintenance.
1. A true test failure will not be detected when run on Travis, thus opening the possibility that broken code could be merged into `master` (unless a reviewer takes the time to check out the PR branch and run the tests locally in their headed environment).  The test failures will most likely not be discovered until the next local build by a developer.

I opened this issue so we can discuss possible alternatives.  Below are three proposals that completely or partially address the previously-enumerated items.  Please feel free to suggest alternatives that I haven't considered.

#### A. Run xvfb on Travis during the entire build

Running xvfb on a headless system is the traditional way to provide a virtual headed environment.  This allows tests that make use of the Swing graphics subsystem to run without throwing `HeadlessException`.

One drawback of this solution is that it allows devs to get lazy when writing unit tests and Swing dependencies may creep in, something that should probably be avoided in the context of a unit test.

#### B. Run xvfb on Travis only during the `integTest` task

Same as (A), but xvfb would be started immediately before running the `integTest` task and stopped immediately after.

This is slightly better than (A) in that unit tests will continue to throw `HeadlessException` if a dev accidentally introduces a Swing dependency.  However, such tests now have to be moved to the `integ_test` folder.  For the types of tests introduced in #2137, that may seem like overkill, as it could split otherwise related tests across multiple files.

#### C. Use JUnit `Assume` instead of handling `HeadlessException`

This solution completely addresses item (1), but only partially addresses item (2).  In this case, we would use the JUnit `Assume` methods to disable headed tests when they are run in a headless environment.  For example, the previously-cited test would become:

```java
@Test(expected = NullPointerException.class)
public void contentsIsRequired() {
  assumeFalse("requires headed environment", GraphicsEnvironment.isHeadless());

  JDialogBuilder.builder()
      .title("title")
      .parentFrame(new JFrame())
      .build();
}
```

I believe this test is cleaner because there is no distracting try-catch block, and the assumption of a headed environment is stated clearly at the beginning of the test.

Even though the test will not be run on Travis, it will be marked as skipped instead of passing.  If we ever have a dashboard of test metrics for the CI builds, it will highlight those tests as not being run rather than hiding them as possible false positives.

**Comment by DanVanAtta** [comment]
Thanks for listing the problem and options clearly - I'm good with `C`

**Comment by ssoloff** [comment]
(C) sounds good to me.  I'll submit a PR soon.

**Comment by ssoloff** [comment]
Resolved by #2176.
